### ISSUE TITLE
Lite ONNX-Inference Version of Ultralytics Without Torch Dependency

### ISSUE THREAD

**Issue by Flippchen** [ultralytics/ultralytics#20676]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Currently, installing the `ultralytics` package automatically pulls in `torch` as a dependency, even when the one only intends to use ONNX models for inference. Since Ultralytics already has pre- and post-processing implemented for ONNX export and inference, it would be beneficial to provide a lightweight version (e.g.`ultralytics[lite]`) that omits the torch dependency entirely. This version would rely only on packages such as numpy for inference-related operations.

### Use case

This feature would reduce the installation size and resource requirements for users deploying ONNX models on edge devices, lightweight servers, or in containerized environments where Torch is unnecessary.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Flippchen, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

🤖 This is an automated response. An Ultralytics engineer will review your feature request and assist you soon!

**Comment by glenn-jocher** [comment]
Hi @Flippchen, thanks for this thoughtful feature request! A lightweight `ultralytics[lite]` package for ONNX inference without the torch dependency would indeed be valuable for edge deployments and containerized environments.

Currently, you can work around this by using `pip install ultralytics --no-deps` and then manually installing only the minimal dependencies needed for ONNX inference (like numpy, opencv, etc.), as outlined in our [custom installation methods](https://docs.ultralytics.com/quickstart/#custom-installation-methods).

We'll consider adding this as a proper installation option in a future release. This is a reasonable request that aligns with our goals of making YOLO models more accessible for deployment across different environments.

**Comment by glenn-jocher** [comment]
You're right to question this, @Flippchen. The current implementation does still import torch in various places throughout the codebase, even in the ONNX inference path. While theoretically ONNX inference should only need onnxruntime and supporting packages, our implementation currently has torch imports in several utility functions used during the inference pipeline.

This is actually why a proper `ultralytics[lite]` would be valuable - we'd need to refactor parts of the codebase to provide torch-free alternatives for these utility functions when in ONNX inference mode.

Thanks for pointing this out - it confirms the value of your feature request. Until we implement this properly, you might need to experiment with mocking the torch imports or creating a minimal custom inference script using onnxruntime directly with your exported models.
